contrib/babelfishpg_tsql/src/iterative_exec.c

```diff
@@ -1542,6 +1571,17 @@ void process_explain(PLtsql_execstate *estate)
 	if (*pltsql_protocol_plugin_ptr && (*pltsql_protocol_plugin_ptr)->stmt_beg)
 		((*pltsql_protocol_plugin_ptr)->stmt_beg) (estate, NULL);
 
+	/* If use_db changed db during the query, return it back to the starting database */
```

Comment:
We should not allow to change db. Let's think about the way to prevent this instead of returning back to the starting db.

Reply:
I think that would be a pretty substantial departure from SQL Server.  SQL Server seems to internally process use db.  Here's a snippet showing use db from sql server

```
    1> create database showplan
    2> go
    1> use showplan
    2> create table test_showplan_all(a int)
    3> go
    Changed database context to 'showplan'.
    1> use master
    2> go
    Changed database context to 'master'.
    1> set showplan_all on
    2> go
    1> use showplan
    2> select * from test_showplan_all
    3> go
    StmtText                                                        StmtId      NodeId      Parent      PhysicalOp                     LogicalOp                      Argument                                       DefinedValues                             EstimateRows   EstimateIO     EstimateCPU    AvgRowSize  TotalSubtreeCost OutputList                                Warnings Type                                                             Parallel EstimateExecutions
    --------------------------------------------------------------- ----------- ----------- ----------- ------------------------------ ------------------------------ ---------------------------------------------- 
    ----------------------------------------- -------------- -------------- -------------- ----------- ---------------- ----------------------------------------- -------- ---------------------------------------------------------------- -------- ------------------
    use showplan                                                              1           1           0 NULL                           NULL                           1                                              NULL                                                NULL           NULL           NULL        NULL             NULL NULL                                      NULL     USE DATABASE                                                            0               NULL
    select * from test_showplan_all                                           2           2           0 NULL                           NULL                           2                                              NULL                                                 1.0           NULL           NULL        NULL        0.0032831 NULL                                      NULL     SELECT                                                                  0               NULL
      |--Table Scan(OBJECT:([showplan].[dbo].[test_showplan_all]))            2           3           2 Table Scan                     Table Scan                     OBJECT:([showplan].[dbo].[test_showplan_all])  [showplan].[dbo].[test_showplan_all].[a]             1.0   3.2035001E-3   7.9600002E-5          11        0.0032831 [showplan].[dbo].[test_showplan_all].[a]  NULL     PLAN_ROW                                                                0                1.0
     
    (3 rows affected)
    1> select * from test_showplan_all
    2> go
    Msg 208, Level 16, State 1, Server EC2AMAZ-536AJ70, Line 1
    Invalid object name 'test_showplan_all'.
````